localization/ar_tag_based_localizer/src/ar_tag_based_localizer_core.cpp

```diff
@@ -268,6 +279,40 @@ void ArTagBasedLocalizer::publish_pose_as_base_link(
   pose_with_covariance_stamped.header.frame_id = "map";
   pose_with_covariance_stamped.pose.pose = tf2::toMsg(map_to_base_link);
 
+  // If latest_ekf_pose_ is older than <ekf_time_tolerance_> seconds compared to current frame, it
+  // will not be published.
+  const double curr_sec = msg.header.stamp.sec + msg.header.stamp.nanosec * 1e-9;
+  const double ekf_sec =
+    latest_ekf_pose_.header.stamp.sec + latest_ekf_pose_.header.stamp.nanosec * 1e-9;
+  if (ekf_sec + ekf_time_tolerance_ < curr_sec) {
```

Comment:
I guess `double` type may not accurately store this value. Using `rclcpp::Time` and `rclcpp::Duration` might be better. 
```suggestion
  const rclcpp::Duration tolerance{
    static_cast<int32_t>(ekf_time_tolerance_),
    static_cast<uint32_t>(ekf_time_tolerance_ - std::floor(ekf_time_tolerance_))};
  if (rclcpp::Time(latest_ekf_pose_.header.stamp) + tolerance < rclcpp::Time(msg.header.stamp)) {
```

Reply:
Thank you! I didn't know there was a class for time management. I agree that it is better to use this.
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4946/commits/3c8fb046b1cec1ff87e74bbfaa3bc0937b5678a0

Reply:
I'm sorry. I suggested wrong code. :bow: Since the second argument of rclcpp::Duration is nanoseconds, we need to pass a number multiplied by 10^9. 
```cpp
  const rclcpp::Duration tolerance{
    static_cast<int32_t>(ekf_time_tolerance_),
    static_cast<uint32_t>((ekf_time_tolerance_ - std::floor(ekf_time_tolerance_)) * 1e9)};
```

Reply:
Indeed, that is so true. I had missed it.

Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4946/commits/da1f56483b1c972d418512fb87d18cc57f021776